docs/tasks/administer-cluster/kubeadm-upgrade-ha.md

```diff
@@ -0,0 +1,179 @@
+# for the versions stick to kubeadm's output (see above)
+$ yum install -y kubelet-<NEW-K8S-VERSION> kubectl-<NEW-K8S-VERSION> kubeadm-<NEW-K8S-VERSION> kubernetes-cni-<NEW-CNI-VERSION>
+$ systemctl start `echo keepalived kubelet docker | tac`
+$ kubectl uncordon <MASTER-NODE-FQDN>
+```
+
+In this example an _rpm_-based system is assumed and `yum` is used for installing the upgraded software. On _deb_-based systems it will be `apt-get update` and then `apt-get install <PACKAGE>=<NEW-K8S-VERSION>` for all packages.
+
+Because on the worker nodes there is no keepalived to stop and start and all processes need to be moved to other nodes cleanly before shutting anything down, `drain` is used instead of `cordon`. In this process all containers will get moved from one node to another at least once. If some of them keep local data you will need some strategy for this scenario. This really depends on the application(s) running in the cluster, and there is no simple, generic solution to this.
+
+On the master (or wherever you run `kubectl` commands):
+
+```shell
+$ kubectl drain --delete-local-data --ignore-daemonsets <WORKER-NODE-FQDN>
+```
+
+On the respective worker node:
+
+```shell
+$ systemctl stop keepalived kubelet docker
+# use your distro's package manager, e.g. 'yum' on RH-based systems
+# for the versions stick to kubeadm's output (see above)
+$ yum install -y kubelet-<NEW-K8S-VERSION> kubectl-<NEW-K8S-VERSION> kubeadm-<NEW-K8S-VERSION> kubernetes-cni-<NEW-CNI-VERSION>
+$ systemctl start `echo keepalived kubelet docker | tac`
+```
+
+Like above, for _deb_-based systems use `apt-get install <PACKAGE>=version` to upgrade the packages to the respective desired versions.
+
+Now the new version of the `kubelet` should be running on the host. Verify this using the following command on the respective host:
+
+```shell
+systemctl status kubelet
+```
+
+If everything was successful, again on the master (or wherever you run `kubectl` commands):
+
+```shell
+$ kubectl uncordon <WORKER-NODE-FQDN>
+```
+
+After all hosts in your cluster have been upgraded, verify that all nodes are available again by executing the following from wherever you run `kubectl` commands:
+
+```shell
+kubectl get nodes
+```
+
+If the `STATUS` column of the above command shows `Ready` for all of your hosts, you are done.
+
+## If something goes wrong
+
+If the upgrade fails the situation afterwards depends on the phase in which things went wrong:
+
+1. If `/tmp/kubeadm upgrade apply` failed to upgrade the cluster it will try to perform a rollback. Hence if that happened on the first master, chances are pretty good that the cluster is still intact.
+
+   You can run `/tmp/kubeadm upgrade apply` again as it is idempotent and should eventually make sure the actual state is the desired state you are declaring. You can use `/tmp/kubeadm upgrade apply` to change a running cluster with `x.x.x --> x.x.x` with `--force`, which can be used to recover from a bad state.
```

Comment:
It's not clear to me under what types of situations a user would be required to use `--force`, I think it should be made a bit more clear to avoid the potential for users using `--force` in a situation where they shouldn't.

Reply:
That's copy and paste from the [(non-HA) `kubeadm` upgrade guide ](https://kubernetes.io/docs/tasks/administer-cluster/kubeadm-upgrade-1-9/). I'd be fine to remove it, but it would probably be better to clarify this in all the `kubeadm` upgrade guides, which I'd consider out of scope for this PR.